### ISSUE TITLE
`rustc -Cprefer-dynamic=yes -Cprefer-dynamic=no` equivalent to `rustc -Cprefer-dynamic=yes`

### ISSUE THREAD

**Issue by shaybarak** [rust-lang/rust#92809]
The `prefer-dynamic` flag has the values "yes" and "no", or if unspecified then it's implicitly "yes".

It's often expected to be able to invoke a tool such as rustc with multiple conflicting values, where it's expected for the last value given to take effect. This can be useful for instance in build systems where a common rule defines default flags to pass to rustc, and then some users add additional flags that may override previously-given flags.

I noticed that the following invocation:
```
rustc -Cprefer-dynamic=yes -Cprefer-dynamic=no
```

Is equivalent to:
```
rustc -Cprefer-dynamic=yes
```

In other words it seems that the first value of `prefer-dynamic` wins, whereas I'd expect the last value to win.
It would be nice to fix this. Thanks!

**Comment by ehuss** [comment]
rustc should be using the last option on the command line.  Can you provide some more details on why it appears that it is using an earlier one?

```
~/Temp> rustc -Cprefer-dynamic=yes -Cprefer-dynamic=no -
fn main() {}
~/Temp> ldd rust_out
        linux-vdso.so.1 (0x00007ffecc5f7000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f7dfb1e9000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f7dfb1c6000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f7dfb1c0000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f7dfafce000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f7dfb259000)
~/Temp> rustc -Cprefer-dynamic=yes -
fn main() {}
~/Temp> ldd rust_out
        linux-vdso.so.1 (0x00007ffc60bd4000)
        libstd-13e2ae73269b4206.so => not found
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fccbf6b5000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fccbf4c3000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fccbf6e0000)
```

**Comment by shaybarak** [comment]
You are correct. I traced my problem to an issue with a rule in the build system that I'm using.
Apologies for the noise and thank you for the help.
